.github/workflows/publish.yml

```diff
@@ -0,0 +1,14 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  call-publish-workflow:
+    uses: Lombiq/NuGet-Publishing-GitHub-Actions/.github/workflows/publish.yml@v1
+    with:
+      source: https://nuget.cloudsmith.io/lombiq/Helpful-Libraries/v3/index.json
+    secrets:
+      apikey: ${{ secrets.LOMBIQ_CLOUDSMITH_API_KEY }}

```

Comment:
Need to add this secret, probably to your org, before publishing

Reply:
Here it is accessing the org secrets and then they are passed as an argument.

Don't think we can access them directly in the call-publish, think we need them as an argument.

(which will annoyingly mean updating each repo, when you want to move it to the different source)

Let me see if there's some way of specifying a default arg in the call-publish-workflow, so it's only overwritten if not supplied.

Reply:
That would be great. So the point is really to make these configs in each repo as lean as possible and cover the common case of publishing to the same Lombiq NuGet (which is what we'll use in the overwhelming majority) with defaults in the Actions repo. (My problem is less about copying things over and over, though that's painful as well, but more about long-term maintainability.)

Reply:
The secrets will have to be passed as an argument (as I thought about it I realized that otherwise someone else could use the workflow externally and then access your secret if it was hard coded into the call-workflow.

I can probably set a default value for the nuget url though, so we don’t need to pass that

Reply:
Do you mean that somebody could refer to the NuGet Actions repo's action in a call-workflow in a repository under another organization, and thus see the secret we refer to under our own org?

Reply:
Yes if GitHub allowed you to access them in that way (not as arguments). 
they don’t which is why we must pass them as an argument

Reply:
Good, because that would be a glaring security issue. However, then I don't understand "I thought about it I realized that otherwise someone else could use the workflow externally and then access your secret if it was hard coded into the call-workflow".

Reply:
Was just illustrating the point about what a glaring security hole it would be.

Reply:
Sorry, I'm not following :). So, can defaults, including for the secret, be in the Actions repo?

Reply:
variables yes, I can default them. secrets no, not as far as I can tell.

it involves parsing the inputs into new variables, so to do so with a secret would not be safe.
eg
```
      - uses: haya14busa/action-cond@v1
        id: sourceeval
        with:
          cond: ${{ inputs.source == '' }}
          # default nuget source. TODO change to nuget.org
          if_true: "https://nuget.cloudsmith.io/lombiq/open-source-orchard-core-extensions/v3/index.json"
          if_false: "${{ inputs.source == '' }}"
```

Reply:
OK, so basically we need to have what's here now, and use global secrets like `LOMBIQ_CLOUDSMITH_API_KEY` (though we'd use one for nuget.org), right? And thus we'd copy this action to the other repos, thus have NuGet publishing configured by default, and if we happen to publish to a different feed in that one project, we'd change the action.

Reply:
This action is actually in the https://github.com/Lombiq/NuGet-Publishing-GitHub-Actions repo, so that's the only one you'll need to update.

I'm going to change the prs on these two repos HE, and HL, to use a secret called `DEFAULT_PUBLISH_API_KEY`.

If you could create that secret, and use your CloudSmith api key initially, then you can update just
- the secret to a nuget apikey, globally against your org
- default NuGet source on NGPGA repo (so many acronyms)

with no changes to HE, and HL

(it was called `LOMBIQ_CLOUDSMITH_API_KEY ` on my forks, so I can differentiate it from my own api keys)

Reply:
I see, I thought that this secret reference needs to be here (i.e. in each repo). Great then!